File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def createKey(self, s):
        key = tuple(sorted(s))
        return key


    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        grp = dict()

        for value in strs:
            key = self.createKey(value)
            if grp.get(key) is None:
                grp[key] = [value]
            else:
                grp[key].append(value)

        return list(grp.values())

File: 0049-group-anagrams/0049-group-anagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */

var createKey = (st) => {
    const key = [...st].sort().join("")
    return key
}

var groupAnagrams = function(strs) {
    const grp = {}
    
    for (const value of strs)
    {
        const key = createKey(value);
        if (!grp[key])
            grp[key] = [value]
        else
            grp[key].push(value);
    }
    return Object.values(grp)
};